iceoryx_hoofs/cmake/IceoryxPackageHelper.cmake

```diff
@@ -111,3 +110,260 @@ Macro(install_package_files_and_export)
     DESTINATION ${DESTINATION_CONFIGDIR}
     )
 endMacro()
+
+Macro(iox_make_unique_includedir)
+    option(MAKE_UNIQUE_INCLUDEDIR
+      "When ON headers are installed to a path ending with folders called \
+      iceoryx/vX.Y.Z/ . This avoids include directory search order issues when \
+      overriding this package from a merged catkin, ament, or colcon workspace."
+      ON)
+
+    if(MAKE_UNIQUE_INCLUDEDIR)
+      set(_unique_dir "iceoryx/v${IOX_VERSION_STRING}")
+      if(PREFIX STREQUAL "")
+        set(PREFIX "${_unique_dir}")
+      else()
+        set(PREFIX "${PREFIX}/${_unique_dir}")
+      endif()
+    endif()
+endMacro()
+
+Macro(iox_set_rpath)
+    set(arguments TARGET)
+    cmake_parse_arguments(IOX "" "${arguments}" "" ${ARGN} )
+
+    if( APPLE )
+        set(IOX_RPATH_PREFIX "@loader_path")
+    elseif ( LINUX OR UNIX )
+        set(IOX_RPATH_PREFIX "\$ORIGIN")
+    endif()
+
+    set(IOX_BUILD_RPATH ${IOX_RPATH_PREFIX}/../iceoryx_hoofs:${IOX_RPATH_PREFIX}/../iceoryx_posh:${IOX_RPATH_PREFIX}/../iceoryx_hoofs/platform:${IOX_RPATH_PREFIX}/../iceoryx_binding_c)
+    set(IOX_INSTALL_RPATH ${IOX_RPATH_PREFIX}/../${CMAKE_INSTALL_LIBDIR})
+
+    # TODO: iox-#1287 to be compatible with our current iceoryx_meta structure where we have build/posh build/hoofs build/binding_c
```

Comment:
Will be addressed in a follow up PR. See todo: `Adjust build destination for all iceoryx modules` in issue #1287